src/vos/vos_query.c

```diff
@@ -200,11 +208,26 @@ query_recx(struct open_query *query, daos_recx_t *recx)
 	recx->rx_idx = entry.en_sel_ext.ex_lo;
 	recx->rx_nr = entry.en_sel_ext.ex_hi - entry.en_sel_ext.ex_lo + 1;
```

Comment:
Is there an advantage to not masking off the parity indicator at this layer ?

Reply:
For EC, it needs to iterate both data space and parity space to know which one is the user wanted recx (MAX or MIN). To hide the details of how to compare/check which one is MAX/MIN (that need some EC detailed logic), in this patch at server-side it iterates both spaces and returns both recxs to client, and client will check which one to return to user.